src/stackdriver-nozzle/stackdriver/metrics_buffer.go

```diff
@@ -16,12 +16,83 @@
 
 package stackdriver
 
-import "reflect"
+import (
+	"context"
+	"github.com/mitchellh/hashstructure"
+	"reflect"
+	"sync"
+	"time"
+)
 
 type MetricsBuffer interface {
 	PostMetric(*Metric)
```

Comment:
Down yonder in [`start`](https://github.com/cloudfoundry-community/stackdriver-tools/blob/521c7641df5c3c767ca994b860752668a832ed3b/src/stackdriver-nozzle/stackdriver/metrics_buffer.go#L82) we're receiving from the context's `Done` channel and flushing metrics there. I think I've got a TODO in `main.go` to give that a few seconds from the time we cancel to complete.